docs/user/get-started/android.md

```diff
@@ -0,0 +1,137 @@
+```
+
+If you are using a `SNAPSHOT` build of `android-components`, it is best to use the master branch, for example:
+
+```Groovy
+apply from: 'https://github.com/mozilla-mobile/android-components/raw/master/components/service/scripts/sdk_generator.gradle'
+```
+
+There are [additional parameters](android-build-configuration-options.md) that can be set to control the behavior of the `sdk_generator.gradle` script, but they are rarely used in normal use.
+
+### Adding new metrics
+
+All metrics that your project collects must be defined in a `metrics.yaml` file.
+This file should be at the root of the module (the same directory as the `build.gradle` file you updated).
+The format of that file is documented [with `glean_parser`](https://mozilla.github.io/glean_parser/metrics-yaml.html).
+To learn more, see [adding new metrics](adding-new-metrics.md).
+
+> **Important**: as stated [before](../adding-glean-to-your-project.md#before-using-glean), any new data collection requires documentation and data-review.
+> This is also required for any new metric automatically collected by the Glean SDK.
+
+### Adding custom pings
+
+Please refer to the [custom pings documentation](../pings/custom.md).
+
+> **Important**: as stated [before](../adding-glean-to-your-project.md#before-using-glean), any new data collection requires documentation and data-review.
+> This is also required for any new metric automatically collected by the Glean SDK.
+
+### Testing metrics
+
+In order to make testing metrics easier 'out of the box', all metrics include a set of test API functions in order to facilitate unit testing.
+These include functions to test whether a value has been stored, and functions to retrieve the stored value for validation.
+For more information, please refer to [Unit testing Glean metrics](../testing-metrics.md).
+
+### Adding metadata about your project to the pipeline
+
+In order for data to be collected from your project, metadata must be added to `probe_scraper`.
+
+These specific steps are described in [the `probe_scraper` documentation](https://github.com/mozilla/probe-scraper#adding-a-new-glean-repository).
+
+## Application-specific steps
+
+The following steps are required for applications using the Glean SDK, but not libraries.
+
+### Initializing the Glean SDK
+
+The Glean SDK should only be initialized from the main application, not individual libraries.  If you are adding Glean support to a library, you can safely skip this section.
+Please also note that the Glean SDK does not support use across multiple processes, and must only be initialized on the application's main process. Initializing in other processes is a no-op.
+Additionally, Glean must be initialized on the main (UI) thread of the applications main process. Failure to do so will throw an `IllegalThreadStateException`.
+
+Before any data collection can take place, the Glean SDK **must** be initialized from the application.
```

Comment:
Maybe out of scope for this bug, but is this _really_ true?  Considering we have the queuing for pre-init tasks, is it worth specifically calling out that we can't do something that we sort of can?